src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1138,14 +1149,13 @@ CHIP_ERROR CASESession::SendSigma3()
     err = mCommissioningHash.Finish(messageDigestSpan);
     SuccessOrExit(err);
 
-    mState = kSentSigma3;
+    mState = State::kSentSigma3;
 
 exit:
 
     if (err != CHIP_NO_ERROR)
     {
         SendStatusReport(mExchangeCtxt, kProtocolCodeInvalidParam);
-        mState = kInitialized;
```

Comment:
Here an in other places where we used to do this, why are we not entering the error state?

Reply:
The full call stack is `CASESession::OnMessageReceived` -> `HandleSigma2_and_SendSigma3` -> `SendSigma3`

The error state is ALWAYS set at the out most function.